core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -411,7 +411,15 @@ class ControllerBrokerRequestBatch(controller: KafkaController) extends  Logging
           .format(broker, stopReplicaWithDelete.mkString(",")))
         debug("The stop replica request (delete = false) sent to broker %d is %s"
           .format(broker, stopReplicaWithoutDelete.mkString(",")))
-        replicaInfoList.foreach { r =>
+
+        val stopReplicaRequestWithDelete = new StopReplicaRequest(controllerId, controllerEpoch, true,
```

Comment:
It may not be a good idea to batch StopReplicaRequests with delete = true since we don't have async delete today, so processing all of these in one request can take a long time.